### DISCUSSION TITLE
Why is the order of the IDs important for TDoA2

### DISCUSSION THREAD

**Discussion by danteppc** [bitcraze/discussions#800] [Q&A] [answered]
Hi!

If I understand correctly, [here](https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/) it says we should use the numbering for anchors as given below:

![](https://www.bitcraze.io/images/tutorials/getting_started_with_lps/loco_ref_system_8_anchors.png)

However, I don't understand **why** it matters. How could this impact the measurements in a way that results in meaningfully different position estimates if lets say I swapped the ids 0 and 2?

**Comment by ataffanel** [comment]
When using TDoA2, packets are received by the Crazyflie in order of the anchor ID and time-difference are pushed in the estimator by received pair. So with TDoA2 the measurements pushed in the estimator would come from anchors 0-1, 1-2, 2-3, etc ... This is why the anchors are placed this way in our example, to maximize the amount of data we get for all 3 dimension for the pairs.

With TDoA3 we still use the last 2 packet to calculate the time difference but packets are sent by the anchor at random which naturally mix the measurements.

**Reply by danteppc** [reply]
I understand that TDoA2 works based on a transmission schedule, but my question is what is the underlying reason that it's important for an anchor to be placed at a relative position to other anchors? For instance, If I only reset the id of anchor 0 to 2 and anchor 2 to 0, like the image below, without changing the coordinates:
![68747470733a2f2f7777772e6269746372617a652e696f2f696d616765732f7475746f7269616c732f67657474696e675f737461727465645f776974685f6c70732f6c6f636f5f7265665f73797374656d5f385f616e63686f72732e706e67-2](https://github.com/bitcraze/discussions/assets/15725082/0380b8ed-1087-4ad0-b553-9957eda8535c)

Is this maybe because the pairs need to be as spread apart as possible? but even then, with a 2 milliseconds transmission gap, how could this be significant when the ToFs are less than 100 nanoseconds?

**Reply by ataffanel** [reply, marked as answer]
Your change should not be noticeable. This is indeed all about the spread of measurement and not about the packet scheduling timing: If we where to organize the anchor naively with 0,1,2,3 on the bottom and 4,5,6,7 on the top, the only measurements that would have some Z component would be [3, 4] and [7, 0]. By organizing the anchor that way we tried to get the measurement spread a little bit more for each consecutive pair.

As a side note: we do not have exact measurements of how much difference this makes for the performance, but we have been setting up our systems that way so we chose to document it accordingly.

**Reply by danteppc** [reply]
Thanks, I actually have the measurements and it made a difference. But didn't know the reason.
